packages/ckeditor5-watchdog/src/editorwatchdog.ts

```diff
@@ -284,14 +306,30 @@ export default class EditorWatchdog<TEditor extends Editor = Editor> extends Wat
 	}
 
 	/**
-	 * Returns the editor data.
+	 * Gets the all data that are required to reinitialize editor instance.
 	 */
-	private _getData(): Record<string, string> {
-		const data: Record<string, string> = {};
+	private _getData(): EditorData {
+		const editor = this.editor!;
+		const roots = editor.model.document.getRootNames();
+		const data: EditorData = {};
+
+		roots.forEach( root => {
+			const markersData = {} as any;
+
+			for ( const marker of editor.model.markers ) {
```

Comment:
This seems incorrect as you will get all markers while you need to only get markers for a given root. Unless you will save data as:

```
{
	roots: {
		rootA: {
			data: ...,
			attributes: ...,
		}, ...
	},
	markers: []
}
```

It seems there's not much difference what will you choose here as this is internal data structure that is used almost right-away. So, I'd propose the one that I proposed :D as it seems it will requre less code (you don't need to filter markers).